packages/graphql-docs-generator/src/index.ts

```diff
@@ -1,112 +1,116 @@
-import * as fs from 'fs';
-import * as path from 'path';
-import * as handlebars from 'handlebars';
-import * as prettier from 'prettier';
-const DEFAULT_MAX_DEPTH = 3;
-
+const handlebars = require('handlebars/dist/handlebars');
 import generateAllOps, { GQLTemplateOp, GQLAllOperations, GQLTemplateFragment, lowerCaseFirstLetter } from './generator';
 import { loadSchema } from './generator/utils/loading';
-const TEMPLATE_DIR = path.resolve(path.join(__dirname, '../templates'));
-const FILE_EXTENSION_MAP = {
-  javascript: 'js',
-  graphql: 'graphql',
-  flow: 'js',
-  typescript: 'ts',
-  angular: 'graphql',
-};
-
-export function generate(
-  schemaPath: string,
-  outputPath: string,
-  options: { separateFiles: boolean; language: string; maxDepth: number },
-): void {
-  const language = options.language || 'graphql';
-  const schemaData = loadSchema(schemaPath);
-  if (!Object.keys(FILE_EXTENSION_MAP).includes(language)) {
-    throw new Error(`Language ${language} not supported`);
-  }
+import { getTemplatePartials, getOperationPartial, getExternalFragmentPartial } from './generator/utils/templates';
+import { getSchemaType } from './generator/utils/getSchemaType';
+
+export { loadSchema } from './generator/utils/loading';
+
+export function generateGraphQLDocuments(
+  schema: string,
+  options: { maxDepth?: number, useExternalFragmentForS3Object?: boolean },
+): GeneratedOperations {
+  const opts = {
+    maxDepth: 2,
+    useExternalFragmentForS3Object: true,
```

Comment:
The `useExternalFragmentForS3Object` is not consistent with the previous code. The current behavior is `default true + option override` where the option is always undefined according to the code block in amplify codegen. The old code uses the `language === 'graphql'` as the condition.

Reply:
That's right. This option needs to be properly set from the top level codegen based on the language, which is work in progress for my upcoming PR. But as far as the docsgen package is concerned, it's going to always return back generated `GraphQL` documents, so the default `true` makes sense here. I have this in my mind.